QModManager/PirateCheck.cs

```diff
@@ -162,28 +175,74 @@ private IEnumerator PlayVideo()
                     Process.Start("https://www.epicgames.com/store/en-US/product/subnautica-below-zero/home");
                     Process.Start("https://discordapp.com/store/skus/535869836748783616/subnautica-below-zero");
                 }
-                yield return new WaitForSeconds(35);
-                Process.Start("https://www.youtube.com/watch?v=dQw4w9WgXcQ");
-                
+                //yield return new WaitForSeconds(35);
+                //Process.Start("https://www.youtube.com/watch?v=dQw4w9WgXcQ");
 
                 while (videoPlayer.isPlaying)
                 {
                     yield return null;
                 }
 
                 yield return StartCoroutine(PlayVideo());
-            }*/
+            }
+
+            private static bool CheckConnection(string hostedURL = "http://www.google.com")
+            {
+                try
+                {
+                    string HtmlText = GetHtmlFromUri(hostedURL);
+                    if (HtmlText == "")
+                        return false;
+                    else
+                        return true;
+                }
+                catch (Exception)
+                {
+                    return false;
+                }
+            }
+            private static string GetHtmlFromUri(string resource)
+            {
+                string html = string.Empty;
+                HttpWebRequest req = (HttpWebRequest)WebRequest.Create(resource);
+                try
+                {
+                    using (HttpWebResponse resp = (HttpWebResponse)req.GetResponse())
+                    {
+                        bool isSuccess = (int)resp.StatusCode < 299 && (int)resp.StatusCode >= 200;
+                        if (isSuccess)
+                        {
+                            using (StreamReader reader = new StreamReader(resp.GetResponseStream()))
+                            {
+                                char[] cs = new char[80];
+                                reader.Read(cs, 0, cs.Length);
+                                foreach (char ch in cs)
+                                {
+                                    html += ch;
+                                }
+                            }
+                        }
+                    }
+                }
+                catch
+                {
+                    return "";
```

Comment:
`string.Empty`

Reply:
So far, `string.Empty` has never been used in this project. I know that there's a small difference between it and `""`, the latter creating an object, but I don't feel like it's necessary to change that.

Reply:
Just trying to get you to pick up some good habits `^_~`

Reply:
Addressed in commit 18f3005